README.md
# monitor.global


changelog.txt
	-- monitor.global v0.5 --

+ добавлен скрипт работы с бэкапами базы данных
+ исправлен баг сортировки по среднему онлайну 
+ исправлена недоработка библиотеки pchart(независимость расстояния подписей к графику от размера графика)
+ добавлены название месяца и период просмотра к графикам за день, неделю, месяц при сравнительном просмотре (title графика)
+ обработан вариант слишком большого title графика на графиках сравнения за неделю и месяц
+ все big графики увеличины до 1200 пикселей в ширину
+ удалены недочеты пропорциональности big графиков
+ добавлена возможность выбора языка
+ весь код переписан под использование языковых файлов
+ добавлены языковые шаблоны
+ доработан sandbox: - невозможность выбора невозможных вариантов :);
                     - добавлен javascript, обрабатывающий данные sandbox;
                     - устранены мелкие баги;
                     - добавлена возможность открывать графики в новом окне.
+ добавлен язык: русский
+ переведена языковая часть библиотеки highslide
+ добавлен функционал для отрисовки flash графиков OFC: - для дня и дня со сравнением
														- для недели и недели со сравнением
														- для месяца и месяца со сравнением
+ добавлены красивые эффекты отрисовки на flash графиках: grow-up, pop-up
+ добавлена возможность выбора flash графиков, для подгрузки flash использован ajax


	-- monitor.global v0.4 --

+ serverdetail.php переработан для работы с AJAX
+ на графики pChart добавлено название системы мониторинга: Monitor Global и версия
+ изменена библиотека pChart, для вывода текста добавлен угол наклона текста
+ доработан вывод времени загрузки и кол-во sql запросов, весь код собран в единую функцию
+ в возможности вывода добавлен вывод данных об онлайне\оффлайне сервера за день\неделю\месяц
+ на графики jpgraph добавлено название системы мониторинга: Monitor Global и версия
+ исправлена проблема отображения: - вертикального текста в графиках jpgraph,
								   - графика при отсутствии данных в jpgraph;
+ скрипт отныне работает из любой директории, локальных привязок нет, только относительные
+ добавлена возможность сравнивать данные об онлайне между разными серверами
+ оптимизировано подключение библиотеки pchart
+ отформатировано положение надписей в jpgraph
+ jpgraph, online\offline: отображение часа в виде 00h, 12h
+ добавлено логгирование в файл: - скрипта сбора данных с серверов и логгирование в файл медленных отрисовок графиков,
								 - доступа к страницам со сбором всех данных о пользователе и парамертах доступа,
								 - ошибок скрипта;
+ в списке серверов (serverlist.php) добавлена возможность сортировки по: алфавиту, среднему онлайну, дате начала записи данных о сервере
+ добавлена обработка ошибок 400, 401, 403, 404 и 500
+ добавлено логгирование на jabber доступа к страницам со сбором всех данных о пользователе и параметрах доступа
+ разработан sandbox


	-- monitor.global v0.3 --
	
+ созданы 
+ улучшен стиль css, добавлены новые графические элементы
+ добавлено время выполнения и кол-во запросов к бд на каждой странице скрипта	
+ созданы: - statistics.php, 
		   - about.php, 
		   - reviewlist.php, 
		   - review.php;
+ добавлена доп. информация о сервере: - max за день\неделю\месяц, average за день\неделю\месяц,
									   - max за все время мониторинга;
+ исправлен баг эффекта opacity для категории
+ добавлено время выполнения и кол-во запросов к бд на каждом отрисованном графике
+ в serverlist.php оптимизированы mysql запросы. вне зависимости от кол-ва данных запросов не больше 3-х
+ исправлена недоработка прорисовки осей графика на pChart
+ изменена библиотека pChart: при построении оси y, max значение по оси y больше в 1.2 раза max графика (чтобы цифры не вылезали за график) 
+ улучшен движок сбора данных с серверов, добавлен контроль над средним и максимальным значением за день
  

	-- monitor.global v0.2 --
	
+ исправлен баг библиотеки pChart (деление на ноль при нулевой разности между max и min)
+ создан движок сайта, система шаблонов, стиль css 
+ созданы: - index.php, 
		   - category.php, 
		   - serverdetail.php, 
		   - serverlist.php;
+ в category.php и serverlist.php добавлены javascript для эффекта opacity
+ в serverdetail.php добавлен javascript для эффекта highslide
+ переписан скрипт javascript календаря
+ добавлена доп. информация о сервере: время старта записи логов для сервера


	-- monitor.global v0.1 --
	
+ создан движок, собирающий данные с серверов
+ создан скрипт тестирования тэгов для данных с сервера
+ написан класс доступа к бд
+ слинкованы скрипты и база данных 
+ создана первая версия statsdaybuilddata.php, скрипта отрисовки графиков

errorpage.php
<?php

/*

========================================================
 Monitor Global
--------------------------------------------------------
 2007-2009, Mike Kozhevnikov
========================================================
 File: errorpage.php
========================================================
 Назначение: Cтраницы с ошибками 400,401,403,404 и 500
========================================================

*/

error_reporting(55);

define ( 'ROOT_DIR', dirname ( __FILE__ ) );

include_once(ROOT_DIR . "/include/config.inc.php");
include_once(ROOT_DIR . "/include/functions.inc.php");
include_once(ROOT_DIR . "/include/class.log.php");

$time_start = getmicrotime();

$page = intval($_GET['page']);
if ($page!=400 && $page!=401 && $page!=403 && $page!=404 && $page!=500)
  {
  $page = NULL;
  exit();
  }

if ($config['accesslogtxt'])
  {
  $logger = new txtLogger("access");
  unset($logger);
  }

$lang = language_cookie(); 

$header = file_get_contents (ROOT_DIR . '/template/' . $lang . '/header.html');
$footer = file_get_contents (ROOT_DIR . '/template/' . $lang . '/footer.html');
$menu   = file_get_contents (ROOT_DIR . '/template/' . $lang . '/menu.html');
include_once(ROOT_DIR . "/language/" . $lang . ".php");

switch ($page)
  {
  case 400:
    $pageinfo=$language['error400title'];
    $pagetext=$language['error400info'];
    break;
  case 401:
    $pageinfo=$language['error401title'];
    $pagetext=$language['error401info'];
    break;
  case 403:
    $pageinfo=$language['error403title'];
    $pagetext=$language['error403info'];
    break;
  case 404:
    $pageinfo=$language['error404title'];
    $pagetext=$language['error404info'];
    break;                                                 
  case 500:
    $pageinfo=$language['error500title'];
    $pagetext=$language['error500info'];
    break;                    
  }

$javascript.= '<script src="/template/js/standart.js" type="text/javascript"></script>';
$title = 'monitor.global - ' . $pageinfo;

$main = '<h1>monitor<span>.global</span> -> ' . $pageinfo . '</h1>
<br><br>' . $pagetext . '<br><br><br><br><br><br><br><br><br><br><br><br><div class="divider"></div>';  

$loadinfo = loadinfo($time_start); 
$footer   = str_replace("{LOADINFO}",$loadinfo['full'],$footer);
$header   = str_replace("{JAVASCRIPT}",$javascript,$header);
$header   = str_replace("{TITLE}",$title,$header);
$output   = $header . $menu . $main . $footer;

echo $output;

?>